### ISSUE TITLE
Bee Textures

### ISSUE THREAD

**Issue by CornyCobly** [Team-Resourceful/ResourcefulBees#11]
It would be a nice feature if you could choose which texture to use for a bee. Consider a world where there a multiple bees with the bee trait "Pigman Bee". It would be nice to distinguish them somehow. The same logic goes for Wither Bees, Creeper Bees, ZomBees. Maybe it is possible to somehow separate the textures from the special traits given to the bees.

**Comment by Epic428** [comment]
I can look into this further and see what I can strum up for it. The main issue right now is that the bee render class checks in a specific order what trait the bee has and then applies the given texture. That is the intended effect. I’m not sure that dynamically loading textures can be done or if it can be that it would be wise to allow it. However, you can use resource packs to override the default supplied textures. 

Considering we only ever register a single entity in the game (currently) and we’re dynamically changing data associated with the spawned in entity there’s only so much we can do, or *should* do. Technically, what we provide with our mod is highly frowned upon by the forge team. Their intention is that any item, block, or entity is static, and every new variant gets registered as a new entity. Obviously, that limits how customizable something can be.

Anyway, I will look into splitting the bee traits and textures apart and try to find a good balance. Any change I make to the structure internally though would cause crashes until the custom bee jsons got updated. So I’d like to do as many core structure updates as I can in a single mod update to limit the frequency of errors from doing so.

And now I’m rambling. Thank you so much for your feedback! We really appreciate it and this is the kind of feedback we’re looking for.

**Comment by CornyCobly** [comment]
I see what you are saying, but the trait skeleton bee does not apply any effects when it attacks a player, it only changes the texture. Also, the nether bee trait makes it so that the bees do not take fire damage without applying a texture. If you split the all textures and the effects that way it would make it even more customizable. I may  be misunderstanding something due to lack of coding and modding experience (I have just recently started learning), but if the current textures are not dynamically loaded, then having more traits like the skeleton bee would not dynamically load textures either. This might have a downside that I am unaware of (more nbt data creates more lag maybe?) but that's how I thought it might be done. 

Basically you could have a tag in the JSON file like "witherBeeTexture": true and another tag "witherBeeEffect": true. This wouldn't change the way things work currently, it would just make it possible for a bee to give you the wither effect without having the wither skeleton texture.

**Comment by Epic428** [comment]
So it seems I may be able to dynamically load textures which gives me an idea. But we will be splitting the traits and the textures up. I’ll work on that next week while I’m on vacation from work. 😁

**Comment by ThatGravyBoat** [comment]
Added in current dev build
